docs/map-api-research.md

```diff
@@ -0,0 +1,115 @@
+
+\*MAU = Monthly Active Users
+
+***
+
+## 2. Here Maps
+
+| API/SDK                         | Description                                                                            | Free Tier (Monthly)    |
+|---------------------------------|----------------------------------------------------------------------------------------|------------------------|
+| **Vector Tile**                 | Pre-rendered tiles & client-side vector rendering                                       | 30 000 requests        |
+| **Geocode / Reverse Geocode**   | Address ↔ geographic coordinate lookup                                                 | 30 000 requests        |
+| **Discover / Search**           | Category & keyword search for places                                                    | 5 000 requests         |
+| **Matrix Routing**              | Time/distance matrices between origins & destinations                                   | 2 500 requests         |
+| **Routing (Car/Bike/Walk)**     | Directions for multiple modes                                                          | 30 000 requests        |
+
+***
+
+## 3. Google Maps Platform
+
+| API                              | Description                                         | Free Tier (Monthly) |
+| -------------------------------- | --------------------------------------------------- | ------------------- |
+| **Maps JavaScript API**          | Interactive, styled maps                            | 10 000 loads        |
+| **Maps Embed API**               | Simple `<iframe>` maps & Street View                | Unlimited           |
+| **Routes: Compute Route Matrix** | Travel-time/distance matrix                         | 10 000 elements     |
+| **Routes: Compute Routes**       | Real-time directions (traffic, transit, bike, walk) | 10 000 requests     |
+| **Geocoding API**                | Forward & reverse geocoding                         | 10 000 requests     |
+| **Geolocation API**              | Device location via cell/WiFi data                  | 10 000 requests     |
+| **Places API — Place Details**   | Detailed info (addresses, opening hours, etc.)      | 10 000 requests     |
+|                                  |                                                     |                     |
+
+***
+
+## 4. OpenStreetMap
+
+- **Data Licensing**: CC BY-SA; free to use, redistribute, and modify with attribution.
+- **Tile Hosting**:
+    - **osm.org tiles**: Public tile server for small usage; heavy use requires self-hosting (e.g., TileServer GL).
+    - **Third-Party Hosts**: Services like Thunderforest, OpenMapTiles (often with their own free tiers).
+- **APIs**:
+    - **Overpass API**: Query raw OSM data (nodes, ways, relations).
+        - ~10,000 requests/day, ~1GB download/day (guidelines); multiple slots per user with queue system.
+    - **Nominatim**: Geocoding/search engine.
+        - **Maximum 1 request per second**; bulk geocoding discouraged.
```

Comment:
"Bulk geocoding is discouraged; for production or high-volume use, consider self-hosting Nominatim."
- should be more appropriate